src/main/java/org/owasp/html/HtmlSanitizer.java

```diff
@@ -229,26 +229,7 @@ private static HtmlStreamEventReceiver initializePolicy(
       Policy policy, HtmlStreamEventProcessor preprocessor) {
     TagBalancingHtmlStreamEventReceiver balancer
         = new TagBalancingHtmlStreamEventReceiver(policy);
-
-    // According to Opera the maximum table nesting depth seen in the wild is
-    // 795, but 99.99% of documents have a table nesting depth of less than 22.
-    // Since each table has a nesting depth of 4 (incl. TBODY), this leads to a
-    // document depth of 90 (incl. HTML & BODY).
-    // Obviously table nesting depth is not the same as whole document depth,
-    // but it is the best proxy I have available.
-    // See http://devfiles.myopera.com/articles/590/maxtabledepth-url.htm for
-    // the original data.
-
-    // Webkit defines the maximum HTML parser tree depth as 512.
-    // http://trac.webkit.org/browser/trunk/Source/WebCore/page/Settings.h#L408
-    // static const unsigned defaultMaximumHTMLParserDOMTreeDepth = 512;
-
-    // The first number gives us a lower bound on the nesting depth we allow,
-    // 90, and the second gives us an upper bound: 512.
-    // We do not want to bump right up against that limit.
-    // 256 is substantially larger than the lower bound and well clear of the
-    // upper bound.
-    balancer.setNestingLimit(256);
+    balancer.setNestingLimit(Integer.MAX_VALUE);
```

Comment:
why max value? can some crafted message blow up the CPU?  
 I think there should be some limit as the comments you are removing  already explained why there is limit. 

in the html you are verifying how deeply it is nested ?

Reply:
@Prashantsurana Sure, please suggest a good limit which we can use. In that html it was 256.

Reply:
After discussion with @dajay0 reduced the value to `1024`.

Reply:
IMO is you read the removed comments is 512 should be fair.

Reply:
Done